jdaviz/configs/specviz/helper.py

```diff
@@ -90,19 +93,26 @@ def get_spectra(self, data_label=None, subset_to_apply=None, apply_slider_redshi
                     if lyr.label == subset_to_apply:
                         spectrum = get_data_method(data_label=lyr.data.label,
                                                    subset_to_apply=subset_to_apply,
-                                                   statistic=statistic)
-                        spectra[lyr.data.label] = spectrum
+                                                   statistic=statistic,
+                                                   cls=Spectrum1D)
+                        # Work around get_data returning Data even with cls specified
```

Comment:
I say we remove this workaround. Otherwise we'll forget it is there and it would make future debugging confusing.

Reply:
I removed the check, but still request `cls=Spectrum1D` from `get_data`.